### DISCUSSION TITLE
How do I get diagnostic info for a crash?

### DISCUSSION THREAD

**Discussion by sicklittlemonkey** [dlemstra/Magick.NET#711] [Help]
Hi, I'm getting a crash in a deployed app (a Windows service) on Windows Server 2012 R2:

Faulting module name: Magick.Native-Q8-x86.dll, version: 7.0.9.0, time stamp: 0x5daf5416
Exception code: 0xc000001d
Faulting module path: C:\Windows\TEMP\Magick.NET.net40.7.14.5.0\Magick.Native-Q8-x86.dll

The code is just:
```
	using (MagickImage image = new MagickImage(imagePath))
	{
		image.Grayscale();
		var ct = image.DetermineColorType();
		image.Write(output);
	}

```

This one machine can reproduce the crash. I can't reproduce it and we've had no similar problems. I'm assuming it might be an OpenCL problem, but I'm not sure how to confirm that. Searching around I found mention of:
%LOCALAPPDATA%\ImageMagick\ImagemagickOpenCLDeviceProfile.xml
... but the support person looking into it couldn't find any such file.

Can I get that info from changing the config files such as log.xml, or is there a way to turn off OpenCL without a code change?

Cheers,
Nick.

**Comment by sicklittlemonkey** [comment]
Hmm, doesn't seem to be an OpenCL problem. Something about the DLL caching is corrupted.

```
Magick.Native-Q8-x86.dll!629b5cd3()
[Frames below may be incorrect and/or missing, no symbols loaded for Magick.Native-Q8-x86.dll]
ntdll.dll!_LdrxCallInitRoutine@16()
ntdll.dll!LdrpCallInitRoutine()
ntdll.dll!LdrpInitializeNode()
ntdll.dll!LdrpInitializeGraph()
ntdll.dll!LdrpPrepareModuleForExecution()
ntdll.dll!LdrpLoadDll()
KERNELBASE.dll!LoadLibraryExW()
[Managed to Native Transition]	
Magick.NET-Q8-AnyCPU.dll!ImageMagick.Environment.NativeEnvironment.SetEnv(string name, string value)
Magick.NET-Q8-AnyCPU.dll!ImageMagick.MagickNET.InitializePrivate(ImageMagick.Configuration.ConfigurationFiles configFiles, string newPath)
Magick.NET-Q8-AnyCPU.dll!ImageMagick.MagickNET.Initialize(ImageMagick.Configuration.ConfigurationFiles configFiles, string path)
Magick.NET-Q8-AnyCPU.dll!ImageMagick.NativeLibraryLoader.ExtractLibrary()
Magick.NET-Q8-AnyCPU.dll!ImageMagick.NativeLibraryLoader.Load()
Magick.NET-Q8-AnyCPU.dll!ImageMagick.Environment.NativeMethods.X86.X86()
[Native to Managed Transition]	
[Managed to Native Transition]	
Magick.NET-Q8-AnyCPU.dll!ImageMagick.Environment.NativeEnvironment.Initialize()
Magick.NET-Q8-AnyCPU.dll!ImageMagick.Environment.Initialize()
Magick.NET-Q8-AnyCPU.dll!ImageMagick.MagickSettings.NativeMagickSettings.NativeMagickSettings()
[Native to Managed Transition]	
[Managed to Native Transition]	
Magick.NET-Q8-AnyCPU.dll!ImageMagick.MagickSettings.MagickSettings()
Magick.NET-Q8-AnyCPU.dll!ImageMagick.MagickImage.MagickImage()
```

**Reply by dlemstra** [reply]
I would be surprised to see an OpenCL issue in a server environment. That will most likely happen on a desktop. From your stack trace I have two guesses:

1. For some reason a 32bit dll is being loaded in a 64bit environment but that doesn't look very likely.
2. The DLL embeded in the `Magick.NET-Q8-AnyCPU.dll` is not written to disk correctly or removed after it has been written. You could try using `MagickAnyCPU.CacheDirectory` to changed that directory.

And If you are only running this on a server I would advise you to switch from the `AnyCPU` version to either the `x86` or the `x64` version of the library.
